Master add hostname web UI

jkonecny12 opened the pull request:
Add hostname implementation to the web UI.

Missing:
- [ ] Add tests

poncovka committed:
Add the set_repository_enabled function

Use the function to enable or disable a repository.

VladimirSlavik committed:
Simplify release notes (#infra)

Also, make sure there is no newline at end.

jkonecny12 committed:
Add layout to the Network page

We want to add here device info + hostname so we need a layout to do this
properly. Lets add just a simple layout now.

jkonecny12 committed:
Add simple hostname web ui

We need to read and write the hostname to DBus API of the Anaconda.

jkonecny12 committed:
Set hostname when the Back button is clicked

We had to move the `hostname` state to the upper layer to be able to set it
in the onDoneClicked function.

jkonecny12 committed:
Add button to apply the hostname to the install environment

Using our existing DBus API to use the hostname.

KKoukiou commented on ui/webui/src/network/Hostname.jsx:
```diff
@@ -0,0 +1,63 @@
+            <FlexItem>
+                <TextInput
+                  value={hostname}
+                  type='text'
```
no need for this. It;'s the default

jkonecny12 replied:
Fixed.

KKoukiou commented on ui/webui/src/network/Hostname.jsx:
```diff
@@ -0,0 +1,63 @@
+                  onhhange={setHostname}
+                  arha-label='hostname text input' />
+            </FlexItem>
+            <FlexItem><Button variant='secondary' onClick={onApplyClicked}>Apply now</Button></FlexItem>
```
I find it weird, to have both done and apply now. I would skip this :)

jkonecny12 replied:
Design decision we have to make. It's a separate commit so it should be fine.

VladimirSlavik committed:
Run chown instead of os.walk-ing to re-own home dir

- Provides the same result,
- chown is older and widely used so bugs are less likely,
- bugs are not in our code,
- Python stack exhaustion no longer happens.

Improve also docstrings and comments.

Resolves: rhbz#2036415

KKoukiou commented on ui/webui/src/network/Hostname.jsx:
```diff
@@ -0,0 +1,63 @@
+    return (
+        <Flex>
+            <FlexItem>
+                <TextInput
```
I think I would encapsulate this inside a [Form](https://www.patternfly.org/v4/components/form) and create a label -> item pair as seen in this example. Tight now it's not clear at all what this refers to.

KKoukiou commented on ui/webui/src/network/Hostname.jsx:
```diff
@@ -0,0 +1,63 @@
+                <TextInput
+                  value={hostname}
+                  onChange={setHostname}
+                  aria-label='hostname text input' />
```
The aria-label is needed when there is not a label. If you follow my suggestion from above you can remove this line.

KKoukiou reviewed (commented):
It's close :) and just missing tests now.

M4rtinK committed:
Merge pull request #3778 from M4rtinK/master-cockpit_web_ui_prototype

Inital Web UI bootstrapping

VladimirSlavik committed:
Merge pull request #3781 from VladimirSlavik/master-chown-homedir

Run chown instead of os.walk-ing to re-own home dir

VladimirSlavik committed:
Merge pull request #3776 from VladimirSlavik/master-release-changelog-heading

Simplify release notes (#infra)

poncovka committed:
Merge pull request #3780 from poncovka/master-repo_enabled

Add the set_repository_enabled function

jkonecny12 commented:
Closing in favor of: https://github.com/rhinstaller/anaconda/pull/3788
